app/scripts/models/auth_brokers/fx-desktop.js

```diff
@@ -93,7 +93,10 @@ define([
       var channel = this._channel || new FxDesktopChannel();
 
       channel.init({
-        window: this.window
+        window: this.window,
+        // The browser is the origin. Events from the browser have no origin.
```

Comment:
For more context - communication with Firefox Desktop for Sync sends [custom events](https://github.com/mozilla/fxa-content-server/blob/c46d67333bd72b74ad3f488c57952635475c7c13/app/scripts/lib/channels/fx-desktop.js#L17) to Firefox, Firefox responds by using [postMessage](https://dxr.mozilla.org/mozilla-central/source/browser/base/content/aboutaccounts/aboutaccounts.js#256). 

The messages that are sent by Firefox have an origin of `null`. This is [known behavior](https://bugzilla.mozilla.org/show_bug.cgi?id=1040257). A quick check shows that at least in Fx 37, message events from Firefox have a field `isTrusted: true`. message events from iframe reliers have `isTrusted: false`. Perhaps this is a viable alternative check.

I will check to see if Fx 29 (the first version of Fx desktop with new Sync) has the same behavior.